### DISCUSSION TITLE
An actual CD cmd.

### DISCUSSION THREAD

**Discussion by rsunde** [go-task/task#1427] [Ideas]
When I discovered Task, I envisioned starting things and moving around, but running CD as a task is pointless due to the sub process thing.

I hope it doesnt go against the philosophy of Task to have a way to change directory as well as start tasks.

I want to be able to group and use task to start things or move somewhere conveniently:

1. task build:myapp
1. task goto:somewhere

imagine somewhere is; ```c:/users/thor/source/repos/work/proofofconcept/net8.0/WinFormsApp1/src/WinFormsApp1.Tests```

my day to day job involves a 800+ projects in a cute little monolith :) (it is what it is) I want to be able to jump around quickly.

**Comment by pd93** [comment]
> When I discovered Task, I envisioned starting things and moving around, but running CD as a task is pointless due to the sub process thing.

> I hope it doesnt go against the philosophy of Task to have a way to change directory as well as start tasks.

Unfortunately, it's not so much that this goes against Task's philosophy as much as it is simply not possible. When you run a program from a shell, it runs as a child process (inheriting the directory from its parent). It is not possible to influence the parent shell in any way (including its current directory) since this would open up serious security issues.

When you run `cd` in a shell, this is actually a special built-in command that will run in the same process as the parent. We are not able to build functionality that does this into an external binary.

This problem is distinct from the fact that each Task command runs in its own subshell which stops you from being able to change directories between Task commands (this is actually fixable).

---

> I want to be able to jump around quickly.

As an aside, this may not be exactly what you're looking for, but if you're a ZSH user like me, then you might find that a [tool like `z`](https://github.com/agkozak/zsh-z) helps you a lot. Since this is written using shell functions, it is able to run in your parent process and change directories for you.



I'm sure there are similar tools out there for Bash/Powershell users too, but I'm not as familiar with those shells and their tools.
